test/core/end2end/run.sh

```diff
@@ -12,8 +13,5 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 set -ex
-export GRPC_POLL_STRATEGY=$1
```

Comment:
It seems that if you introduced an env variable (e.g. `GRPC_TEST_EVENT_ENGINE` since it seems to be only used in tests) that will be used by TestEnvironment, you could have saved yourself (and the reviewer) 300 occurrences of `grpc::testing::TestEnvironment env(&argc, argv);` (and other related edits).

Reply:
I used that approach in an earlier version. Because we support bazel 3.x, we need to use something like a generic `run.sh` (or a genrule, if the script itself were offensive to some sensibilities), so packaging the test execution as a `"<command> [args]"`string  made the most sense. Also, I didn't think ~200 copies of a one-character change would be very onerous to review, apologies if that was annoying to you. Aside from the burden of changing a function signature (which is already done), what arguments do you have that environment variables are a superior approach?